dev/tasks/crossbow.py

```diff
@@ -230,8 +230,26 @@ def github_assets(self, task):
             release = repo.release_from_tag(task.tag)
         except github3.exceptions.NotFoundError:
             return {}
-        else:
-            return {a.name: a for a in release.assets()}
+
+        assets = {a.name: a for a in release.assets()}
+
+        artifacts = {}
+        for artifact in task.artifacts:
+            # artifact can be a regex pattern
+            pattern = re.compile(artifact)
+            matches = list(filter(None, map(pattern.match, assets.keys())))
+            num_matches = len(matches)
+
+            # validate artifact pattern matches single asset
+            if num_matches > 1:
```

Comment:
@cpcloud This is the main downside of using patterns.

Reply:
Does this need to handle the case of `num_matches == 0`? Shouldn't there be at least 1 match?

Reply:
We highlight artifacts with `num_matches == 0` as `MISSING`.